src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -12,37 +17,210 @@ import {
+          (emailAcc, email) => {
+            sources.set('email', [
+              ...(sources.get('email') ?? []),
+              email.source,
+            ]);
+            return emailAcc || !isEditableSource(email.source);
+          },
+          false,
+        );
+        const foundNonEditablePhone = person.phoneNumbers.nodes.reduce(
+          (phoneAcc, phone) => {
+            sources.set('phone', [
+              ...(sources.get('phone') ?? []),
+              phone.source,
+            ]);
+            return phoneAcc || !isEditableSource(phone.source);
+          },
+          false,
+        );
+        return acc || foundNonEditableEmailAddress || foundNonEditablePhone;
+      },
+      false,
+    );
+
+    const uniqueSources: DataInfo['uniqueSources'] = {
+      contact: [],
+      address: [],
+      email: [],
+      phone: [],
+      ...Object.fromEntries(
+        Array.from(sources, ([dataType, ArrOfSources]) => [
+          dataType,
+          [...new Set(ArrOfSources)].map((source) => sourceToStr(t, source)),
+        ]),
+      ),
+    };
+
+    const contactIsNotEditable =
+      isContactNonEditable || isAddressNonEditable || hasNonEditablePersonData;
+
+    return {
+      canDelete: !contactIsNotEditable,
+      uniqueSources,
+    };
+  }, [contactSources]);
+
   return (
     <Modal
       isOpen={open}
       title={t('Delete Contact')}
       handleClose={() => setOpen(false)}
     >
       <DialogContent dividers>
-        <DialogContentText>
-          {t(
-            'Are you sure you want to permanently delete this contact? Doing so will permanently delete this contacts information, as well as task history. This cannot be undone. If you wish to keep this information, you can try hiding this contact instead.',
-          )}
-        </DialogContentText>
+        {dataInfo.canDelete ? (
+          <Typography>
+            {t(
+              `Are you sure you want to permanently delete this contact? Doing so will permanently delete this contacts information, as well as task history. This cannot be undone. If you wish to keep this information, you can try hiding this contact instead.`,
+            )}
+          </Typography>
+        ) : (
+          <>
+            <Typography>
+              {t(
+                `Be cautious when deleting this contact, as its data may sync with Donation Services or other third-party systems. Deleting the contact will not remove it from those systems; consider hiding the contact instead.`,
+              )}
+            </Typography>
+            <Typography fontWeight="bold">
+              {t(
+                `For contacts originating from Siebel or Donor Hub, email Donation Services to request deletion.`,
+              )}
+            </Typography>
+            <br />
+            <br />
+            <Typography variant="h6">{t('Data sources:')}</Typography>
+            <List dense={true}>
+              {!!dataInfo.uniqueSources.contact.length && (
```

Comment:
If you don't use my suggestion above, it looks like `sources.contact` is never pushed to, so `uniqueSources.contact` will always be empty.